### PR TITLE
Integration test improvements for Minikube

### PR DESCRIPTION
**Pull request by ijsnellf**
Several changes I end up adding to the integration tests when I am doing testing in Minikube. If possible, I'd like to merge them.

### CODE REVIEW

File: test/integration/tcp.go [hunk 1]
```diff
@@ -44,6 +44,10 @@ func (t *tcp) run() error {
 	funcs := make(map[string]func() status)
 	for _, src := range testPods {
 		for _, dst := range testPods {
+			if src == dst {
+				continue // minikube bug prevents this scenario from working
```
Comment (rshriram) [inline test/integration/tcp.go]: May be qualify a bit more? Add a -minikube flag to bazel.

File: test/util/kubernetes.go [hunk 2]
```diff
@@ -81,7 +81,7 @@ func GetAppPods(cl kubernetes.Interface, ns string) (map[string][]string, error)
 			return pods, err
 		}
 		items = list.Items
-		ready := true
+		ready := len(items) > 0
```
Comment (rshriram) [inline test/util/kubernetes.go]: This should not be necessary since we now check if the containers are running before declaring the pod ready

Comment (andraxylia) [reply test/util/kubernetes.go]: I also have changes in the same fct in PR, where I check for pods/containers in multiple namespaces. You can revert this.

File: test/integration/tcp.go (hunk 1 above)
Comment (andraxylia) [reply test/integration/tcp.go]: Only the t->t scenario causes issues in minikube, I have this in my PR as well.

**Comment by ijsnellf** [comment]
@andraxylia: since most of these changes are already in your PR, please either close this one, or lgtm it if you want the changes to be in their own PR.

**Pull request closed without merging**